### PR TITLE
Firewalld support

### PR DESCRIPTION
**Pull request by khappone**
Added code so that the role configures firewalld (if present), otherwise
it configures the default iptables rules for forwarding.

### CODE REVIEW

File: tasks/firewalld.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+- name: Check for masquerading
+  shell: firewall-cmd --zone {{ firewalld_masq_zone }} --query-masquerade
+  register: masquerade
+  always_run: yes
+  failed_when: masquerade.rc >= 2
+
+- name: Add masquerading to zone
+  shell: firewall-cmd --zone {{ firewalld_masq_zone }} --add-masquerade --peramanent
```
Comment (rlaurika) [inline tasks/firewalld.yml]: There is a typo here: "--peramanent"

File: tasks/main.yml [hunk 2]
```diff
@@ -2,18 +2,14 @@
...
 
-- name: allow related and established traffic in forwarding chain in iptables
-  command: iptables -I FORWARD 1 -i "{{ external_interface }}" -o "{{ internal_interface }}" -m state --state RELATED,ESTABLISHED -m comment --comment "related and established" -j ACCEPT
-  when: iptablesrules.stdout.find("related and established") == -1
-
-- name: enable masquerading
-  command: iptables -t nat -A POSTROUTING -s "{{ internal_net }}" -o "{{ external_interface }}" -m comment --comment "masquerading for internal net" -j MASQUERADE
-  when: iptablesrules.stdout.find("masquerading for internal net") == -1
+- include: firewalld.yml
+  when: firewalld_state.rc != 1
```
Comment (rlaurika) [inline tasks/main.yml]: If "firewall-cmd --state" were to give some other return code than 0 or 1, then both configuration options iptables and firewalld would be taken, right? That might cause a mess.

**Comment by rlaurika** [comment]
The general layout looks fine, but I'm wondering if all scenarios have been tested? I would assume firewall-cmd would give you a clear error when you pass a flag it doesn't recognize ("--peramanent").

**Comment by khappone** [comment]
Good catch. This problem only comes up with the first run, and we currently have limited options in redeploying these.

**Comment by khappone** [comment]
Updated code to fix typo.

**Comment by khappone** [comment]
Updated logic in main.yml, so that both versions can't run at the same time

**Pull request merged**